### PR TITLE
Bug #9681

### PR DESCRIPTION
**Pull request by kiokoman**
Permit Uppercase Titles but not duplicate

### CODE REVIEW

File: sysutils/pfSense-Status_Monitoring/files/usr/local/www/status_monitoring.php [hunk 1]
```diff
@@ -123,7 +121,7 @@ function createSlug($string) {
 }
 
 //add a new view and make sure the string isn't empty
-if ($_POST['add-view'] && !empty($view_title) && ($view_title != "default")) {
+if ($_POST['add-view'] && !empty($view_title) && !preg_match("/[dD][eE][fF][aA][uU][lL][tT]/",$view_title)) {
```
Comment (jim-p) [inline sysutils/pfSense-Status_Monitoring/files/usr/local/www/status_monitoring.php]: Why go through all that trouble? You could do this with:  `strtolower($view_title) != "default"` or if you really want `preg_match()`, the pattern can be `/default/i`

File: sysutils/pfSense-Status_Monitoring/files/usr/local/www/status_monitoring.php [hunk 2]
```diff
@@ -143,7 +141,7 @@ function createSlug($string) {
 //remove current view
 if ($_POST['remove-view']) {
 
-	if ($view_title == "default") {
+	if (preg_match("/[dD][eE][fF][aA][uU][lL][tT]/",$view_title)) {
```
Comment (jim-p) [inline sysutils/pfSense-Status_Monitoring/files/usr/local/www/status_monitoring.php]: See previous comment

**Comment by jim-p** [comment]
To effectively track changes and bug/feature requests, we require an entry on the [pfSense Redmine issue tracker](https://redmine.pfsense.org) associated with every pull request, and likewise the Redmine entry should also have a link back to the pull request.

Please associate this PR to a Redmine issue either by locating an existing issue at https://redmine.pfsense.org or by creating a new issue. Add a link to the Redmine issue that points to this PR, and also add a link on the PR to the Redmine issue.

For more information, see https://www.netgate.com/docs/pfsense/development/submitting-a-pull-request-via-github.html

Thanks!

**Comment by kiokoman** [comment]
https://redmine.pfsense.org/issues/9681

**Comment by kiokoman** [comment]
https://redmine.pfsense.org/issues/9681

File: sysutils/pfSense-Status_Monitoring/files/usr/local/www/status_monitoring.php [hunk 3]
```diff
@@ -123,7 +121,7 @@ function createSlug($string) {
 }
 
 //add a new view and make sure the string isn't empty
-if ($_POST['add-view'] && !empty($view_title) && ($view_title != "default")) {
+if ($_POST['add-view'] && !empty($view_title) && !preg_match("/default/i",$view_title)) {
```
Comment (rbgarga) [inline sysutils/pfSense-Status_Monitoring/files/usr/local/www/status_monitoring.php]: This change to use preg_match is not the same as previous string comparison. Any string containing the word `default` will match that regular expression. I believe the proper check here would be `(strtolower($view_title) != "default")`

File: sysutils/pfSense-Status_Monitoring/files/usr/local/www/status_monitoring.php [hunk 4]
```diff
@@ -143,7 +141,7 @@ function createSlug($string) {
 //remove current view
 if ($_POST['remove-view']) {
 
-	if ($view_title == "default") {
+	if (preg_match("/default/i",$view_title)) {
```
Comment (rbgarga) [inline sysutils/pfSense-Status_Monitoring/files/usr/local/www/status_monitoring.php]: This change to use preg_match is not the same as previous string comparison. Any string containing the word `default` will match that regular expression. I believe the proper check here would be `(strtolower($view_title) == "default")`

**Comment by kiokoman** [comment]
all done !

**Pull request merged**